### ISSUE TITLE
Incorrect behaviour with files containing `?`

### ISSUE THREAD

**Issue by mtoohey31** [lewis6991/gitsigns.nvim#473]
### Description

When gitsigns is running in a file whose path contains a question mark (`?`), and that file is not yet tracked by git, if there are files whose path is identical to that of the current file, but with the question mark replaced by any character, and one of those files is tracked by git, diff signs will be inacurate. This is very much an edge-case, and it's not a big deal, but I figured I should mention it.

### Neovim version

NVIM v0.6.1

### Operating system and version

5.16-arch

### Expected behavior

No diff signs should be displayed, since the file is untracked, so there's nothing to compare it against.

### Actual behavior

![2022-02-09T23:52:49,117735742-05:00](https://user-images.githubusercontent.com/36740602/153339663-da05a31f-c659-42c6-8738-3e3143a91ce3.png)

I expect this is the result of a shell or git expanding the `?` character to indicate a wildcard, so we probably just need tweak the path escaping. I'm using the fish shell, in case that ends up being related.

### Minimal config

```Lua
vim.o.packpath = '/tmp/nvim/site'

local plugins = {
  gitsigns = 'https://github.com/lewis6991/gitsigns.nvim',
  plenary  = 'https://github.com/nvim-lua/plenary.nvim',
  -- ADD OTHER PLUGINS _NECESSARY_ TO REPRODUCE THE ISSUE
}

for name, url in pairs(plugins) do
  local install_path = '/tmp/nvim/site/pack/test/start/'..name
  if vim.fn.isdirectory(install_path) == 0 then
    vim.fn.system { 'git', 'clone', '--depth=1', url, install_path }
  end
end

require('gitsigns').setup{
  debug_mode = true, -- You must add this to enable debug messages
  -- ADD GITSIGNS CONFIG THAT IS _NECESSARY_ FOR REPRODUCING THE ISSUE
}

-- ADD INIT.LUA SETTINGS THAT IS _NECESSARY_ FOR REPRODUCING THE ISSUE
```

### Steps to reproduce

1. ~~`nvim -nu minimal.lua`~~ (no configuration is necessary, other than having the plugin installed and enabled)
2. `mkdir test && cd test`
3. `git init`
4. `touch 1234`
5. `git add 1234 && git commit`
6. `nvim 12\?4`, then add a line and write the file. Diff signs will be shown, even though the file is untracked.

**Comment by lewis6991** [comment]
Thanks for the report. Will be sure to fix this.

**Comment by lewis6991** [comment]
I cannot reproduce this. Can you provide the output of `Gitsigns debug_messages`?

**Comment by mtoohey31** [comment]
Sure, here's the output:

```
run_job: git --no-pager --version                                                                                                                                                                   
setup_highlights: Highlight GitSignsAdd is already defined
setup_highlights: Highlight GitSignsChange is already defined
setup_highlights: Highlight GitSignsDelete is already defined
setup_highlights: Deriving GitSignsAddNr from GitSignsAdd
setup_highlights: Deriving GitSignsChangeNr from GitSignsChange
setup_highlights: Deriving GitSignsDeleteNr from GitSignsDelete
setup_highlights: Deriving GitSignsAddLn from DiffAdd
setup_highlights: Deriving GitSignsChangeLn from DiffChange
setup_highlights: Deriving GitSignsDeleteLn from DiffDelete
setup_highlights: Deriving GitSignsCurrentLineBlame from NonText
setup_highlights: Deriving GitSignsAddInline from TermCursor
setup_highlights: Deriving GitSignsDeleteInline from TermCursor
setup_highlights: Deriving GitSignsChangeInline from TermCursor
setup_highlights: Deriving GitSignsAddLnInline from GitSignsAddInline
setup_highlights: Deriving GitSignsChangeLnInline from GitSignsChangeInline
setup_highlights: Deriving GitSignsDeleteLnInline from GitSignsDeleteInline
setup_highlights: Deriving GitSignsAddLnVirtLn from GitSignsAddLn
setup_highlights: Deriving GitSignsChangeVirtLn from GitSignsChangeLn
setup_highlights: Deriving GitSignsDeleteVirtLn from GitSignsDeleteLn
setup_highlights: Deriving GitSignsAddLnVirtLnInLine from GitSignsAddLnInline
setup_highlights: Deriving GitSignsChangeVirtLnInLine from GitSignsChangeLnInline
setup_highlights: Deriving GitSignsDeleteVirtLnInLine from GitSignsDeleteLnInline
attach(1): Attaching (trigger=setup)
run_job: git --no-pager config user.name
run_job: git --no-pager rev-parse --show-toplevel --absolute-git-dir --abbrev-ref HEAD
run_job: git --no-pager --git-dir=[REDACTED]/.git -c core.quotepath=off ls-files --stage --others --exclude-standard --eol [REDACTED]/notes/ws?.md
watch_gitdir(1): Watching git dir
run_job: git --no-pager --git-dir=[REDACTED]/.git show :0:notes/ws4.md
update(1): updates: 1, jobs: 5
run_job: git --no-pager rev-parse --show-toplevel --absolute-git-dir --abbrev-ref HEAD
```

**Comment by lewis6991** [comment]
I'm still not able to reproduce this, which suggest this may be a platform or build specific issue, however I've found out about a `--literal-pathspecs` option which may work. I've added this in #527. @mtoohey31 are you able to test it?

**Comment by mtoohey31** [comment]
Yep, that fixes it! ONce that's merged this can be closed.

**Comment by lewis6991** [comment]
Closed with d0b42c0fc7a3d6c27a3cfaf959cb59a949388633
